core/geometry/src/curve/internalContexts/CurveCurveIntersectXY.ts

```diff
@@ -1072,6 +1079,10 @@ export class CurveCurveIntersectXY extends RecurseToCurvesGeometryHandler {
       this.dispatchLineStringBSplineCurve(
         lsA, this._extendA0, this._extendA1, this._geometryB, this._extendB0, this._extendB1, false,
       );
+    } else if (this._geometryB instanceof TransitionSpiral3d) {
+      const linestring = LineString3d.create();
+      this._geometryB.emitStrokes(linestring);
+      this.computeLineStringLineString(lsA, linestring, false);
```

Comment:
Note that this method doesn't pass in extend flags. In order to disallow extension of the spiral approximant, you can add extend flags to the `computeLineStringLineString` arg list (like `computeSegmentLineString`) and refactor.